Chunking

jsarabia opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bible-translation-tools/orature/587)
<!-- Reviewable:end -->

jsarabia committed:
wip

jsarabia committed:
Copy metacontent for not versebyverse, ignore links

jsarabia committed:
Temporary fix for verse total for pluginviewmodel

jsarabia committed:
Add chunks working

Export not working, Chunks can be added, recorded against

Need to fix: chunk count, source text- they are blocking gets, need a better solution

jsarabia committed:
WIP

jsarabia committed:
Update ChapterPageViewModel.kt

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Load Chunking Wizard

jsarabia committed:
Update ChapterPage.kt

jsarabia committed:
add chunks from chunking wizard

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Fix next/previous chunk

jsarabia committed:
Reset Chunking for new chapter

jsarabia committed:
Ignore duplicates after chunking

jsarabia committed:
UI

jsarabia committed:
Save chunk markers to derived source audio

jsarabia committed:
Use chunked audio as source (hacky)

jsarabia committed:
Add back chapter text

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Update ChapterPage.kt

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Add verse by verse chunking

jsarabia committed:
Update VerseByVerseChunking.kt

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Find verses overlapping chunks (getting text broken)

jsarabia committed:
Update VerseByVerseChunking.kt

jsarabia committed:
chunk text

jsarabia committed:
Update build.yml

jsarabia committed:
ignore tests

jsarabia committed:
Ignore more tests

jsarabia committed:
Update build.yml

jsarabia committed:
more ignores

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Remove use of ContentLabel.of()

ContentLabel does not allow for discerning between Chunk and Verse and as such cannot be used accurately.

BookElements, Collections, and Content already have label fields anyway, use that.

jsarabia committed:
Merge branch 'jbs-labels' into jbs-chunk-storage

jsarabia committed:
Update VerseByVerseChunking.kt

jsarabia committed:
Fix chunk number displays

jsarabia committed:
Clear chunks when switching chapters

jsarabia committed:
Change insert to List

Changes inserting chunks to a list to begin adding draft number, as a first step towards resetting chunking

jsarabia committed:
Fix mp3 chunked audio support

jsarabia committed:
export chunked audio (#579)

* Move source audio to project

* Name chunked audio based on source

* Copy source audio on export

jsarabia committed:
reset chunks (#582)

* Reset Chunks

* Collect by draft number

* Delete files when resetting chapter

* Update AudioPluginViewModel.kt

* Add source content back

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Write chunk file

jsarabia committed:
delete chunk file on reset

AnonymousWalker committed:
Chapter page - Chunk wizard (#584)

* working

* clean up

jsarabia committed:
refactor import (#586)

* Get Chapters of Project without takes

* Import verse by verse where content exists

* Mostly importing chunks and verses

* Update ProjectImporter.kt

* Update WorkbookRepository.kt

* verse count

jsarabia committed:
Update ChapterPage.kt

jsarabia committed:
Update ProjectFilesAccessor.kt

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Ignore tests

jsarabia committed:
WIP

jsarabia committed:
Update ProjectFilesAccessor.kt

jsarabia committed:
cleanup

jsarabia committed:
Delete Translation Dialog (#589)

jsarabia committed:
delete css fix

jsarabia committed:
Update chunking-wizard.css

jsarabia committed:
Bugfix: Finding appropriate source audio file

previously the cue file could be selected as source audio, resulting in a crash

jsarabia committed:
Wrong source audio chunk selected on recording page

Many chunk can share the same starting verse number, and so the sort needs to be used.

jsarabia committed:
Clear Connections before leaving chunking wizard

jsarabia committed:
Free audio resources in chunking wizard

Frees audio resources so wav files can be written to in Windows.

AnonymousWalker commented:
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/Chapter.kt` line 35 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/587#-N4bkYm0BJib5W-DRZJV:-N4bkYm19qMcB4Lm1xa4:bpbj9h2) ([raw file](https://github.com/bible-translation-tools/orature/blob/ce6a6a78dbbcb6c5aa4ecb095fbb6e1ddc260200/common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/Chapter.kt#L35)):*
> ```Kotlin
> import org.wycliffeassociates.otter.common.domain.content.Recordable
> 
> data class Chapter(
> ```

this should be a class, as it should have been. Also, could the constructor parameters (chunks, chunkCount, addChunk, reset) go to another class /wrapper? I see a lot specific pieces that are not strongly related to a Chapter other than chunks.


<!-- Sent from Reviewable.io -->

AnonymousWalker commented:
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/Chunk.kt` line 68 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/587#-N4blWzw-S9UurvjqQAa:-N4blWzw-S9UurvjqQAb:b-9cnsog) ([raw file](https://github.com/bible-translation-tools/orature/blob/ce6a6a78dbbcb6c5aa4ecb095fbb6e1ddc260200/common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/Chunk.kt#L68)):*
> ```Kotlin
>         if (start != other.start) return false
>         if (end != other.end) return false
>         if (draftNumber != other.draftNumber) return false
> ```

you may consider using `when { }`


<!-- Sent from Reviewable.io -->

AnonymousWalker commented:
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/FileNamer.kt` line 77 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/587#-N4bmHkQ6jVXCnIMB5iW:-N4bmHkQ6jVXCnIMB5iX:b-atfndc) ([raw file](https://github.com/bible-translation-tools/orature/blob/ce6a6a78dbbcb6c5aa4ecb095fbb6e1ddc260200/common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/FileNamer.kt#L77)):*
> ```Kotlin
>         val verseNum = when (start) {
>             null -> null
>             else -> verseFormat.format(sort)
> ```

How about this?
`val verseNum = start?.let { verseFormat.format(sort) }`


<!-- Sent from Reviewable.io -->

AnonymousWalker commented:
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/VerseByVerseChunking.kt` line 35 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/587#-N4bwjl69EvB3QEFzovu:-N4bwjl69EvB3QEFzovv:b-63j9r3) ([raw file](https://github.com/bible-translation-tools/orature/blob/ce6a6a78dbbcb6c5aa4ecb095fbb6e1ddc260200/common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/VerseByVerseChunking.kt#L35)):*
> ```Kotlin
> 
>     init {
>         accessor = ProjectFilesAccessor(
> ```

Could we pass a ProjectFilesAccessor from constructor instead of re-creating a new one here? 


<!-- Sent from Reviewable.io -->

AnonymousWalker reviewed (changes requested):
> Reviewed 33 of 59 files at r1, 28 of 29 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/587)* status: 61 of 69 files reviewed, 7 unresolved discussions (waiting on @AnonymousWalker and @jsarabia)



<!-- Sent from Reviewable.io -->

AnonymousWalker committed:
Fix verse highlight in Record scripture page (#606)

* fix verse marker count

* refactor

AnonymousWalker committed:
add filter when deleting take & content

AnonymousWalker committed:
working draft number

AnonymousWalker committed:
working clean up for chunks relay (from db)

AnonymousWalker committed:
remove draft variable

AnonymousWalker committed:
update navigation between chunks

AnonymousWalker committed:
fix takes disappear when reset chapter and create take

AnonymousWalker committed:
remove old code

AnonymousWalker committed:
refactor

AnonymousWalker committed:
remove max draft fetching

AnonymousWalker committed:
remove filter take for numbering

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage-rebase

jsarabia committed:
Add chunk file to backup project

jsarabia committed:
Merge pull request #614 from Bible-Translation-Tools/jbs-chunk-storage-rebase

chunk storage rebase

jsarabia committed:
Merge branch 'jbs-chunk-storage' into aw-fix-delete

jsarabia committed:
Merge pull request #608 from Bible-Translation-Tools/aw-fix-delete

Fix reset chapter

jsarabia committed:
refactor markers (#612)

* Move track control inside waveform control

* Fit marker track to end

* Update TimecodeHolder.kt

* wip

* wip

* wip

* Revert "wip"

This reverts commit 6607ce236aedf33067d431bc480e9632fb7ce798.

* Fix widths

* Update MarkerPlacementWaveform.kt

* center images

* Update WaveformFrame.kt

* Update WaveformFrame.kt

* Update WaveformFrame.kt

* Revert "Update WaveformFrame.kt"

This reverts commit 571077639659549866c3e8e6549bbe78fbf13616.

* Revert "Update WaveformFrame.kt"

This reverts commit 3c809e5cc63f945b7c24c586494797094adac6df.

* Make MarkerTrackControl a Layout rather than a Control

* Scrolling Waveforms as Layouts not Controls

* Remove need for runlater block

* Update VerseMarkerModel.kt

* Fix marker refreshing

* Fix layout, scale image to exclude tracks

* Set min height for responsiveness

* Fix highlight color order

* update minimap markers

* Operations Stack

* Update VerseMarkerModel.kt

* Undo/Redo for Verse Marker App

Still need some changes to Chunk Page

* Fix markers for chunk page

* Move code for markers into an interface

* Update VerseMarkerModel.kt

jsarabia committed:
Fix Begin/Continue translation

Setting the working chunk is not updated when the user creates a verse by verse translation as it was only done on dock.

Chunking will correctly result in begin translation because of the view re-docking.

While executing this in a map is not ideal, the change to a relay means this no longer completes.

AnonymousWalker committed:
style chunking controls

jsarabia committed:
Fix navigate back

jsarabia committed:
observable waveform builder (#619)

* WIP

* fix disposal concurrent access exception

* Delete RxTest.kt

* Remove WaveformImageBuilder

* Update VerseMarkerViewModel.kt

* optimizations

* Cleanup

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Fix tests (#623)

* Update WorkbookDataStoreTest.kt

* Update WorkbookDataStoreTest.kt

* Fix WorkbookDataStore tests

Accessing source text is now more complex as text reads from the USFM.

This functionality should be moved elsewhere, and not be part of the WorkbookDataStore.

As such, for now, the tests have been removed.

* Fix AudioExporterTest

* Fix RecordScriptureViewModelTest

jsarabia committed:
Get tests compiling again (#624)

* Update WorkbookDataStoreTest.kt

* Update WorkbookDataStoreTest.kt

* Fix WorkbookDataStore tests

Accessing source text is now more complex as text reads from the USFM.

This functionality should be moved elsewhere, and not be part of the WorkbookDataStore.

As such, for now, the tests have been removed.

* Fix AudioExporterTest

* Fix RecordScriptureViewModelTest

* Fix WorkbookPageViewModelTest

* Fix ResourceListViewModelTest

* Uncomment TestProjectExport

* Backup stuck

Project backup does not halt

* Test Project Import running

* TestProjectCreate Running

* Fix TestInitializeUlb

* TestInitializeProjects running

jsarabia committed:
Jbs fix tests chunking (#625)

* Update WorkbookDataStoreTest.kt

* Update WorkbookDataStoreTest.kt

* Fix WorkbookDataStore tests

Accessing source text is now more complex as text reads from the USFM.

This functionality should be moved elsewhere, and not be part of the WorkbookDataStore.

As such, for now, the tests have been removed.

* Fix AudioExporterTest

* Fix RecordScriptureViewModelTest

* Fix WorkbookPageViewModelTest

* Fix ResourceListViewModelTest

* Uncomment TestProjectExport

* Backup stuck

Project backup does not halt

* Test Project Import running

* TestProjectCreate Running

* Fix TestInitializeUlb

* TestInitializeProjects running

* TestWorkbookRepository compiling

jsarabia committed:
Fix more tests (#629)

* Update WorkbookDataStoreTest.kt

* Update WorkbookDataStoreTest.kt

* Fix WorkbookDataStore tests

Accessing source text is now more complex as text reads from the USFM.

This functionality should be moved elsewhere, and not be part of the WorkbookDataStore.

As such, for now, the tests have been removed.

* Fix AudioExporterTest

* Fix RecordScriptureViewModelTest

* Fix WorkbookPageViewModelTest

* Fix ResourceListViewModelTest

* Uncomment TestProjectExport

* Backup stuck

Project backup does not halt

* Test Project Import running

* TestProjectCreate Running

* Fix TestInitializeUlb

* TestInitializeProjects running

* TestWorkbookRepository compiling

* Fix get takes for collection sql

The order of this left join resulted in getting takes of other collections beyond the one intended

* Update TakeDao.kt

* Initialize imported draftNumber to 1

imported draftNumbers should be 1; if no takes are found, the content entities will be purged anyway.

* Update TestProjectImport.kt

jsarabia commented on common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/Chapter.kt:
```diff
@@ -32,26 +38,48 @@ data class Chapter(
+                .forEach { draft.add(it) }
+            while (draft.size != chunkTotal) {
+                sleep(50)
+            }
```
explain why this is necessary

jsarabia committed:
Update WorkbookFileNamerBuilder.kt

jsarabia committed:
Update DatabaseMigrator.kt

jsarabia committed:
Sets the default draft number to 1

The default draft number is 1 so that migrated databases will appropriately have their content marked as draft 1

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Fix merge issue

jsarabia committed:
Update playback controls ui in marker app

jsarabia committed:
Fix WorkbookRepo Tests and lazy load chunks

As Chunks are now a Relay and not an Observable, some changes had to be made in the WorkbookRepositoryTest to reflect the fact that now blocking get cannot be called and chunks are constructed in a different way.

Additionally, to make them lazily evaluated like they were previously, the function call to set up the relay is wrapped in a lazy block, (and the type is updated in the Chapter constructor parameters as well).

jsarabia committed:
Update TestWorkbookRepository.kt

jsarabia committed:
Additional test corrections

chunks are now lazy,
more chunk counts updated to reflect chapters not being chunked at import time.

jsarabia committed:
Comment out resource related tests

jsarabia committed:
Update DatabaseMigrator.kt

jsarabia committed:
Fix checking for compilability

jsarabia committed:
Update ChunkingViewModel.kt

jsarabia committed:
Update ChunkingViewModel.kt

jsarabia committed:
Remove some debugging print statements

jsarabia committed:
Update WorkbookDataStore.kt

jsarabia committed:
Update ProjectFilesAccessor.kt

jsarabia committed:
Implement equals/hashcode for chapter

jsarabia committed:
Rename Chunking Usecase

jsarabia committed:
Clarify verseByVerse option in project creation

jsarabia committed:
Add logging around plugins

jsarabia committed:
Update AudioPlugin.kt

jsarabia committed:
Update tornadofx

Updates tornadofx to include a fix for Fragments leaking

jsarabia committed:
Revert "Update tornadofx"

This reverts commit 94f072bb99867d8df32f1fd692c4d142bed4789c.

jsarabia committed:
Update Chapter.kt

jsarabia committed:
Update tornadofx (second try)

Required updating bindings for now ambiguous parameters.

jsarabia committed:
Update AudioPlugin.kt

jsarabia committed:
Rename chunk page

jsarabia committed:
Remove Project Source Audio when resetting a draft

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Update ChapterPageViewModel.kt

jsarabia committed:
Include source audio when importing resumable projects

jsarabia committed:
Fix source audio bug

Chapter 1 would match with chapter 10 audio files.

jsarabia committed:
Update ProjectFilesAccessor.kt

jsarabia committed:
Image Cleanup

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

AnonymousWalker committed:
CSS Fix for chunking (#638)

* refactor to universal button style

* styling effect for playback buttons

AnonymousWalker committed:
Pause audio on undock - chunking steps (#639)

* invoke pause() when leaving the view

* pause playback in verbalize

* fix crash on playback when returning to verbalize

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

AnonymousWalker reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/587)* status: 17 of 112 files reviewed, 9 unresolved discussions (waiting on @AnonymousWalker and @jsarabia)

___
*[`jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/waveform/IMarkerViewModel.kt` line 14 at r14](https://reviewable.io/reviews/bible-translation-tools/orature/587#-N96vIXM9gxZU_ii1tNB:-N96vIXM9gxZU_ii1tNC:bd7i6b7) ([raw file](https://github.com/bible-translation-tools/orature/blob/4e648ca7a65c05f9e61e981ad13b817fac5c90b1/jvm/controls/src/main/kotlin/org/wycliffeassociates/otter/jvm/controls/waveform/IMarkerViewModel.kt#L14)):*
> ```Kotlin
> import org.wycliffeassociates.otter.jvm.controls.model.VerseMarkerModel
> 
> interface IMarkerViewModel {
> ```

Just a reminder, this default implemented interface needs documentation to avoid confusion with its purpose.


<!-- Sent from Reviewable.io -->

jsarabia committed:
Fix marker app memory leaks (#647)

* Move lambdas to properties

* Fix Memory leaks in marker app

Lambdas with view model calls can leak without being held in properties.

Timer moved to the viewmodel instead of the view so that a timer tick does not restore the viewmodel to the components map after the scope is deregistered.

All avenues to close the marker app are now done with firing an event so that the event is properly cleaned up.

jsarabia committed:
Merge dev into jbs-chunk-storage

jsarabia committed:
Fix AudioPlayerController issue

Using composite disposable did not update the slider or button on the simpleaudioplayer

jsarabia committed:
Fix button icon binding

jsarabia committed:
Update Messages_en.properties

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

AnonymousWalker committed:
Fix crash deleting chunking project (#648)

* nullify audio player when leaving page

* rename

* Update AudioPlayerController.kt

Co-authored-by: Joe <jsarabia1247@gmail.com>

jsarabia committed:
Add chunk relay to disposables (#650)

* Add chunk relay to disposables

Adds the chunk relay in the workbook repository to the disposables map so it is cleaned up when the workbook is closed.

jsarabia committed:
Extract ChunkAudioUseCase

jsarabia committed:
Merge branch 'jbs-chunk-storage' of https://github.com/Bible-Translation-Tools/Orature into jbs-chunk-storage

jsarabia committed:
Move Timer (#652)

Moves the timer from the view to viewmodel for the purposes of preventing memory leaks

jsarabia committed:
Scope chunking wizard to clean memory (#654)

* Scope chunking wizard to clean memory

Cleans the state and releases memory by using scopes which can deregister when the user leaves the chunking area.

Updates tornadofx to use a fix of the Wizard which does not add the wizard to the onCloseRequest of the stage (leaks memory)

Adds a breadcrumb for the chunking wizard.

* Update ChapterPage.kt

mXaln reviewed (commented):
> Reviewed 1 of 59 files at r1, 1 of 29 files at r3, 5 of 13 files at r5, 13 of 29 files at r6, 10 of 32 files at r8, 14 of 26 files at r9, 9 of 18 files at r10, 1 of 16 files at r11, 14 of 24 files at r12, 8 of 11 files at r13, 16 of 20 files at r15, 3 of 3 files at r16, 2 of 2 files at r17, 1 of 2 files at r18, 3 of 3 files at r19, 5 of 5 files at r20, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/587)* status: all files reviewed, 11 unresolved discussions (waiting on @AnonymousWalker, @jsarabia, and @mXaln)

___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/Chapter.kt` line 41 at r9](https://reviewable.io/reviews/bible-translation-tools/orature/587#-N7LNfUP7rfd30M8RnWU:-N7LNfUP7rfd30M8RnWV:b-t2vvlz) ([raw file](https://github.com/bible-translation-tools/orature/blob/783b6c03810feba9b0aa0e4ff95ecb3158bef298/common/src/main/kotlin/org/wycliffeassociates/otter/common/data/workbook/Chapter.kt#L41)):*
> ```Kotlin
>     override val resources: List<ResourceGroup>,
>     override val subtreeResources: List<ResourceMetadata>,
>     private val lazychunks: Lazy<ReplayRelay<Chunk>>,
> ```

Maybe make it camelCase?
___
*[`common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/WorkbookRepository.kt` line 231 at r19](https://reviewable.io/reviews/bible-translation-tools/orature/587#-N9aAro9C91r36qVscSO:-N9aAro9C91r36qVscSP:b-ahstsa) ([raw file](https://github.com/bible-translation-tools/orature/blob/71d30ccfeec623ab4edc8eda2be4ec39bec491cf/common/src/main/kotlin/org/wycliffeassociates/otter/common/persistence/repositories/WorkbookRepository.kt#L231)):*
> ```Kotlin
>     ): ReplayRelay<Chunk> {
>         val rr = ReplayRelay.create<Chunk>()
>         db.getContentByCollectionActiveConnection(chapterCollection)
> ```

This name is too long and I don't quite understand what Active Connection means.


<!-- Sent from Reviewable.io -->

jsarabia committed:
Fix memory leaks on recording page (#657)

AudioPlayerController is not released from the SourceContentFragment, causing a leak when the RecorderView returns.

Also modified exiting to always use the events so the unsubscribe is properly called.

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Add toString to book (#658)

Allows logging books

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

mXaln committed:
Restore the accessability feature of the waveform (#667)

* Added alternative narration view for recording page

* Accessibility on chunking wizard

AnonymousWalker committed:
Fix DB init tests (#668)

* update test db version

* clean up

* use max int value

jsarabia committed:
Create chunk markers rather than repurpose verse markers (#666)

* Namespace chunk markers

Adds chunk markers as "orature-chunk-#" rather than repurposing verse markers, and thus allowing both to coexist.

* Merge Media when importing Resumable Project

Changes the Resumable Project Import to always import sources (which would merge source audio).

* Upload Test for SourceAudioFile

* Update SourceAudioFile.kt

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Update AudioPlayerController.kt

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

AnonymousWalker committed:
remove logging

AnonymousWalker committed:
Insert content derivative when creating chunks/verses (#676)

* insert into content derivative when adding chunks

* clean up and use completable method

* capture null/empty from repo getter()

* add test

* add test for content repo - linkDerivedToSource()

* comment

AnonymousWalker committed:
No highlight for chunk (verse) page (#675)

* avoid highlight in chunk/verse text

reset verse count if docked page is not chapter and reset highlight when changing to next/previous verse/chunk

* add NO_HIGHLIGHT_INDEX constant

AnonymousWalker committed:
remove verbalize - UI + ViewModel (#679)

AnonymousWalker committed:
Test Database - Daos (#670)

* squash commits

* add .close() database

* refactor & cleanup test

Add description & documentation; include record count before/after an operation.

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

jsarabia committed:
Merge branch 'dev' into jbs-chunk-storage

mXaln committed:
Chunk storage tests (#656)

* Added ChunkAudioUseCaseTest

* Added tempdir

* moved test data to utils

* more test refactoring

* viewmodels refactoring

* updated utils

* added CreateChunks Tests

* added ResetChunks Tests

* added one more assert

* Update TestProjectExport.kt

Co-authored-by: Joe <jsarabia1247@gmail.com>

jsarabia committed:
Fix edge cases for chunking (#692)

Handles the case where there are no verse markers associated with the source audio as well as if the start verse marker is not at the beginning of the file.

jsarabia committed:
Noop commit to trigger CI/CD

AnonymousWalker committed:
Retain chunks created in chunks.json when import (#701)

* only resets chapters that do not have content (chunks not defined)

* refactor json serializer to class

* refactor lines

* update chunk file when resetting a chapter

* refactor

* rename
